ibis/bigquery/client.py

```diff
@@ -285,17 +319,26 @@ class BigQueryClient(SQLClient):
     dialect = comp.BigQueryDialect
 
     def __init__(self, project_id, dataset_id):
-        self.client = bq.Client(project_id)
-        self.dataset_id = dataset_id
+        """
+        Parameters
+        ----------
+        project_id : str
+            A project name
+        dataset_id : str
+            A ``<project_id>.<dataset_id>`` string or just a dataset name
+        """
+        (self.data_project_id,
+         self.run_project_id,
+         self.dataset_id) = parse_project_and_dataset(project_id, dataset_id)
+        self.client = bq.Client(self.data_project_id)
```

Comment:
Nit: The `google-cloud-python` team considers [`project` a keyword parameter to Client](https://googlecloudplatform.github.io/google-cloud-python/latest/bigquery/reference.html#google.cloud.bigquery.client.Client) and doesn't make guarantees on the positional order.

Also, shouldn't this be the `run_project_id`? I guess either way we'll have to pass an explicit project somewhere. I just tend to think of the one on the client as the one I'm getting charged to.

Reply:
> Nit: The google-cloud-python team considers project a keyword parameter to Client and doesn't make guarantees on the positional order.

Thanks for letting me know, I'll update that.

The reason I chose that is because we're using `self.client` everywhere to get table names and schema information. Ultimatley I only care about the project getting billed just before execution, and everything else is related to the project I'm getting data from. Does that make sense?

Reply:
Makes sense, yeah. Only one place where we run queries whereas there are many places where we want the default project for the dataset.

Reply:
It would be a breaking change, but we could just require passing in the project to `execute` instead of requiring it to be part of the constructor.

Reply:
Actually, I don't think it would be a breaking change. Let me see what I can do here.

Reply:
I tinkered with this a bit, and I actually like the way it is now, so that users don't have to pass in a `project` keyword on every execution. We could have a default, but that brings additional complexity in the form of defaults, since we'd want to have a way to make it easy to execute if you know your billing project.

Reply:
Per: https://github.com/ibis-project/ibis/issues/1429

I think rather than using the default project on `bigquery.Client` at all, Ibis should explicit about which project it is using (data project vs billing project). This extra explicitness is still hidden from the user, but it'll be a bit easier to reason about and also handle cross-project queries.